Archives/VolFile.h

```diff
@@ -53,11 +53,20 @@ namespace Archives
 			CompressionType compressionType;
 		};
 
+		// Specify boundary padding for a volume file section
+		enum class VolPadding : uint32_t
+		{
+			TwoByte = 0,
+			FourByte = 1 // Default value
+		};
+
 		struct SectionHeader
 		{
+			SectionHeader();
+			SectionHeader(std::array<char, 4> tag, uint32_t length, VolPadding padding = VolPadding::FourByte);
 			std::array<char, 4> tag;
 			uint32_t length : 31;
-			uint32_t padding : 1; // 0 = 4-byte boundary padding, 1 = 2-byte boundary padding
+			VolPadding padding : 1;
```

Comment:
You may want to verify `sizeof(SectionHeader)` here. I believe in some cases, changing the type can break apart bit fields into separate storage locations. In particular, `uint32_t` is not `VolPadding`. It is close though, in that it's an `enum` based on `uint32_t`, so it might still work.

I'd recommend adding a `static_assert` check to ensure the size of the struct is and always remains the size required by the file format:
```
static_assert(sizeof(SectionHeader) == 8, "SectionHeader not of required size");
```

Reply:
This aspect of bit fields is certainly beyond my knowledge of the language. I placed the static_assert in the constructor of VolFile. It didn't give me any errors, so I guess on MSVC it is okay. 

Let me know if it is more appropriate to place it somewhere else though. I've never actually used static_assert before.

Reply:
I would place it right after the struct definition. It is a compile time check, so it doesn't need to be placed within a function.